emcc.py

```diff
@@ -2517,6 +2517,12 @@ def phase_linker_setup(options, state, newargs):
   settings.SUPPORTS_PROMISE_ANY = feature_matrix.caniuse(feature_matrix.Feature.PROMISE_ANY)
   if not settings.BULK_MEMORY:
     settings.BULK_MEMORY = feature_matrix.caniuse(feature_matrix.Feature.BULK_MEMORY)
+    if settings.MEMORY64 and settings.MIN_NODE_VERSION < 180000:
+      logger.warning(
+        "Disabling bulk memory because it doesn't work correctly with wasm64 in Node.js < 18.\n"
+        "Set MIN_NODE_VERSION to 180000 or above to enable it."
+      )
+      settings.BULK_MEMORY = 0
```

Comment:
Do we disable a feature like this in any other cases?

Is there a reason not to do this:
```diff
diff --git a/tools/feature_matrix.py b/tools/feature_matrix.py
index 7aebf78c5..438fae430 100644
--- a/tools/feature_matrix.py
+++ b/tools/feature_matrix.py
@@ -40,10 +40,11 @@ min_browser_versions = {
   },
   Feature.BULK_MEMORY: {
     'chrome': 75,
     'firefox': 79,
     'safari': 150000,
+    'node': 18000,
   },
   Feature.MUTABLE_GLOBALS: {
     'chrome': 74,
     'firefox': 61,
     'safari': 120000,
```
?

Reply:
The issue is that bulk memory is supported.. just not under wasm64.. and the feature matrix can't yet express such things.

Reply:
Yeah this. I mean, we could also declare that wasm64 is only supported on canary Node (20+) but that seems like a bigger change and looks like harder to do on CI given that you currently need to list all tests that depend on canary manually in CircleCI config.

Reply:
I see, thanks. Maybe add a comment explaining this, to avoid confusion when reading the code?

Reply:
Hm I thought the warning message is self-describing / acts as a comment?

Reply:
Sorry, to be clear, what I mean is the specific issue here that is rather unique. Say:
```
# Note that we do not update tools/feature_matrix.py for this, as this issue is
# specific to wasm64. That is, the required version of node for bulk memory
# is very different for wasm32 and wasm64, and the feature matrix atm does
# differentiate them. Another option would be to use the least common
# denominator, but that would be limiting for wasm32, so instead we special-
# case this situation here.
```

Reply:
Okay, np.

Reply:
Added a variation of your comment with some rewording / more details.

Reply:
Thanks, I think this is a lot clearer and years from now someone may read it and be grateful :smile: